Extensions/Guardrail+Settings.swift

```diff
@@ -138,12 +138,11 @@ public extension Guardrail where Value == HKQuantity {
     static func maximumBolus(supportedBolusVolumes: [Double]) -> Guardrail {
         let maxBolusThresholdUnits: Double = 30
         let maxBolusWarningThresholdUnits: Double = 20
-        let minimumSupportedBolusVolume = supportedBolusVolumes.first!
-        let recommendedLowerBound = minimumSupportedBolusVolume == 0 ? supportedBolusVolumes.dropFirst().first! : minimumSupportedBolusVolume
+        let supportedBolusVolumes = supportedBolusVolumes.drop { $0 == 0 }
         let recommendedUpperBound = min(maxBolusWarningThresholdUnits, supportedBolusVolumes.last!)
         return Guardrail(
             absoluteBounds: (supportedBolusVolumes.first!...supportedBolusVolumes.last!).clamped(to: 0...maxBolusThresholdUnits),
```

Comment:
Doesn't this clamping potentially result in an unsupported bolus volume? Perhaps the best choice is to drop all from supportedBolusVolumes that are <= 0 or > maxBolusThresholdUnits and then simply take first and last?

Reply:
Ah!  You are right.  Will fix.